src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -1532,6 +1741,356 @@ internal C(char c)
+            comp.VerifyDiagnostics(
+                // (9,12): warning CS8618: Non-nullable property 'Prop' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+                //     public C1() // 1
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C1").WithArguments("property", "Prop").WithLocation(9, 12));
+        }
+
+        [Fact]
+        [WorkItem(41110, "https://github.com/dotnet/roslyn/issues/41110")]
+        public void TrackMemberStateAcrossInitializers()
+        {
+            var source = @"
+class C
+{
+  static string? P1 { get; set; } = """";
+
+  static string P2 { get; set; } = P1;
+  static string f1 = P1;
+}
+
+class D
+{
+  static string? f1 = """";
+
+  static string f2 = f1;
+  static string P1 { get; set; } = f1;
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(44180, "https://github.com/dotnet/roslyn/issues/44180")]
+        public void MemberNotNull_PropertiesFields()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+class Property
+{
+  public string P { get; set; }
+
+  public Property() { Init(); P.ToString(); }
+
+  [MemberNotNull(nameof(P))]  void Init() => P = """";
+}
+
+class Field
+{
+  public string F;
+
+  public Field() { Init(); F.ToString(); }
+
+  [MemberNotNull(nameof(F))] void Init() => F = """";
+}";
+            var comp = CreateCompilation(new[] { source, MemberNotNullAttributeDefinition }, options: WithNonNullTypesTrue(), parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(41296, "https://github.com/dotnet/roslyn/issues/41296")]
+        public void InitializeInTryAndInCatch()
+        {
+            var source = @"
+using System;
+public class C
+{
+    string field;
+
+    public C()
+    {
+        try
+        {
+            M2(out field);
+        }
+        catch (Exception)
+        {
+           if (field is null)
+           {
+              field  =  """";
+           }
+        }
+    }
+
+    static void M2(out string s) => throw null!;
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(44212, "https://github.com/dotnet/roslyn/issues/44212")]
+        public void InitializeUsingNullCoalescingAssignment()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+class C
+{
+    string f;
+
+    public C()
+    {
+        Prop ??= """";
+        f.ToString();
+    }
+
+    public C(byte b)
+    {
+        if (Prop == null)
+        {
+            Prop = """";
+        }
+        f.ToString();
+    }
+
+    [MemberNotNull(nameof(f))]
```

Comment:
>[MemberNotNull(nameof(f))] [](start = 4, length = 26)

Does this apply to both accessors? If so, are the calls to the setter above important, or would `if (Prop == null) { }` be sufficient to initialize `f`?

Reply:
I think it does apply to both accessors. This test was taken from a bug report, and I think it was originally intended to demonstrate the incongruity between MemberNotNull analysis and uninitialized fields analysis.

It might make sense to edit out or split the MemberNotNull related parts, since the point now is to demonstrate that non-nullable fields can be initialized using `if (field is null) field = value;` (which didn't work before this change) or `field ??= value`.